dotnet/tests/MicroFocus.FAS.AdapterSdkSchema.Tests/AdapterSdkSchemaObjectBuilderTest.cs

```diff
@@ -0,0 +1,333 @@
+/**
+ * Copyright 2022-2023 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using System.Collections;
+using Xunit;
+
+namespace MicroFocus.FAS.AdapterSdkSchema.Tests
+{
+    public class AdapterSdkSchemaObjectBuilderTest
+    {
+        [Fact]
+        public void TestCreateDataSetWithSchemaObjectBuilder()
+        {
+            Console.WriteLine("TestCreateDataSetWithSchemaObjectBuilder...");
+            var document = new Dictionary<string, object>();
+            AdapterSdkSchemaObjectBuilder documentBuilder = new(new DictionarySchemaObjectBuilder(document));
+            // Integer
+            documentBuilder.SetColumnCount(3);
+            // Long
+            documentBuilder.SetRootFileSize(418);
+            // Boolean
+            documentBuilder.SetHasAttachments(false);
+            // Datetime
+            documentBuilder.SetDateArchived(DateTime.Now);
+            // Single-valued string
+            documentBuilder.SetAddressBcc("jdoe@gmail.com");
+            // Multi-valued string
+            documentBuilder.SetAddressTo("person1@gmail.com", "person2@there.com");
+
+            // Multi-valued Nested object, Json encoded
+            documentBuilder.SetAccounts(
+                new List<Action<AdapterSdkSchemaObjectBuilder.AccountsObjectBuilder>> {
```

Comment:
We should look at adding a set overload that takes a parameter array of actions, to simplify this usage (i.e. to remove the need to construct this list).  We can either merge this and then do a separate PR for it, or just continue with this one.

Reply:
Added setter that takes an array